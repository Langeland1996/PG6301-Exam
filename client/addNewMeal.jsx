import React, {useState} from "react";

function FormInput({ label, value, onChangeValue }) {
    return (
        <div>
            <label>
                <strong>{label}</strong>{" "}
                <input value={value} onChange={(e) => onChangeValue(e.target.value)} />
            </label>
        </div>
    );
}

export function AddNewMeal({ createMeal }) {
    const [dish, setDish] = useState("");
    const [ingredients, setIngredients] = useState("");
    const [allergies, setAllergies] = useState("");
    const [vegan, setVegan] = useState("");
    const [vegetarian, setVegetarian] = useState("");

    function handleSubmit(event) {
        event.preventDefault();
        createMeal({ dish, ingredients, allergies, vegan, vegetarian });
    }

    return (
        <form onSubmit={handleSubmit}>
            <h1>Add new movie</h1>

            <FormInput label={"dish:"} value={dish} onChangeValue={setDish} />
            <FormInput label={"Ingredient:"} value={ingredients} onChangeValue={setIngredients} />
            <FormInput label={"Allergies"} value={allergies} onChangeValue={setAllergies} />
            <FormInput label={"Vegan"} value={vegan} onChangeValue={setVegan} />
            <FormInput label={"Vegetarian"} value={vegetarian} onChangeValue={setVegetarian} />
            <button>Submit</button>
        </form>
    );
}