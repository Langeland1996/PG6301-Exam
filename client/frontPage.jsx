import {Link} from "react-router-dom";
import React from "react";

export function FrontPage() {
    return (
        <div>
            <h1>Welcome to food catering</h1>
            <h4>Click on the menu and choose between a variety of foods</h4>
            <ul>
                <>
                    <Link to={"/menu"}>Menu</Link>
                    <br></br>
                    <br></br>
                    Admin only: <Link to={"/addNewMeal"}> Add meal to menu</Link>
                    <br></br>
                    <br></br>
                    <Link to={"/"}>Log out</Link>
                </>
            </ul>
        </div>
    );
}