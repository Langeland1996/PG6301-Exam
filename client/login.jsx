import { Link } from "react-router-dom";
import React, { useEffect } from "react";

export function Login() {
  useEffect(() => {
    window.location.href =
      "https://accounts.google.com/.well-known/openid-configuration";
  }, []);

  return <div>Login updated!</div>;
}
