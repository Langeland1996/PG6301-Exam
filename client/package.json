{
  "name": "client",
  "version": "1.0.0",
  "description": "",
  "scripts": {
    "test": "jest",
    "dev": "parcel watch index.html",
    "build": "npm install --include=dev parcel build index.html"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "devDependencies": {
    "@babel/preset-env": "^7.20.2",
    "@babel/preset-react": "^7.18.6",
    "jest": "^29.3.1",
    "jest-babel": "^1.0.1",
    "jest-environment-jsdom": "^29.3.1",
    "parcel": "^2.8.0",
    "process": "^0.11.10"
  },
  "dependencies": {
    "@parcel/transformer-react-refresh-wrap": "^2.8.0",
    "@testing-library/react": "^13.4.0",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-router-dom": "^6.4.3"
  },
  "babel": {
    "presets": [
      "@babel/preset-react",
      [
        "@babel/preset-env",
        {
          "targets": {
            "node": "current"
          }
        }
      ]
    ]
  },
  "jest": {
    "testEnvironment": "jsdom",
    "collectCoverageFrom": [
      "**/*.jsx",
      "**/*.js",
      "!dist/**",
      "!coverage/**"
    ]
  }
}
