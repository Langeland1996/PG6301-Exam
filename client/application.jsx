import {FetchJSON} from "./Middleware/fetchJSON";
import {BrowserRouter, Route, Routes} from "react-router-dom";
import {FrontPage} from "./frontpage.jsx";
import {ListMenu} from "./Menu/listMenu";
import {ListVeganMenu} from "./Menu/listVeganMenu";
import {ListVegetarianMenu} from "./Menu/listVegetarianMenu";
import React from "react";
import {Login} from "./login";
import {AddNewMeal} from "./addNewMeal";
import {MainPage} from "./mainPage";

export function Application() {
    async function listMenu() {
        return await FetchJSON("/api/menu");
    }

    async function listMenuOnlyVegan() {
        return await FetchJSON("/api/menu/vegan");
    }

    async function listMenuOnlyVegetarian() {
        return await FetchJSON("/api/menu/vegetarian");
    }

    function createMeal({dish, ingredients, allergies, vegan, vegetarian }){
        if (dish === "" || ingredients === "" || allergies === "" || vegan === "" || vegetarian === "") {
            return <div>Input invalid</div>
        }


        const jsonData = {
            Dish: dish,
            Ingredients: ingredients,
            Allergies: allergies,
            Vegan: vegan,
            Vegetarian: vegetarian
        }

        const requestOptions = {
            method: 'POST',
            headers: { 'Content-Type': 'application/json' },
            body: JSON.stringify(jsonData)
        };

        fetch("/api/menu" , requestOptions)
            .then(response => response.json())
            .then(data => this.setState({ postId: data.id }));
    }

    return (
        <BrowserRouter>
            <Routes>
                <Route path={"/"} element={<FrontPage/>}/>
                <Route path={"/login"} element={<Login/>}/>
                <Route path={"/mainPage"} element={<MainPage/>}/>
                <Route path={"/menu"} element={<ListMenu listMenu={listMenu}/>}/>
                <Route
                    path={"/menu/vegan"}
                    element={<ListVeganMenu listMenuOnlyVegan={listMenuOnlyVegan}/>}
                />
                <Route
                    path={"/menu/vegetarian"}
                    element={<ListVegetarianMenu listMenuOnlyVegetarian={listMenuOnlyVegetarian}/>}
                />
                <Route path={"/addNewMeal"} element={<AddNewMeal createMeal={createMeal}/>}/>
            </Routes>
        </BrowserRouter>
    );
}