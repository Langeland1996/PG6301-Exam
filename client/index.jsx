import React from "react";
import ReactDOM from "react-dom";
import { Application } from "./application";
import { BrowserRouter as Router } from "react-router-dom";

ReactDOM.render(
  <React.StrictMode>
      <Application />
  </React.StrictMode>,
  document.getElementById("app")
);
