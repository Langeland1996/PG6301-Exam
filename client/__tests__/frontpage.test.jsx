import { createRoot } from "react-dom/client";
import { act } from "@testing-library/react";
import React from "react";
import { FrontPage } from "../frontpage.jsx";
import { BrowserRouter } from "react-router-dom";

describe("Application component", () => {
  it("Check Application Snapshot", async () => {
    const element = document.createElement("div");
    await act(async () => {
      createRoot(element).render(
        <BrowserRouter>
          <FrontPage />
        </BrowserRouter>
      );
    });
    expect(element.innerHTML).toMatchSnapshot();
  });
});
