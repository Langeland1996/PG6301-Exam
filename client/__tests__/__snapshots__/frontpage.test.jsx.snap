// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application component Check Application Snapshot 1`] = `"<div><h1>Login</h1><ul><li><a href="/Login">Click to login</a></li><li><a href="/mainPage">Go to website</a></li></ul></div>"`;
